### DISCUSSION TITLE
Building Iterations of gpt-engineer's creation?

### DISCUSSION THREAD

**Discussion by ghost** [AntonOsika/gpt-engineer#264] [Q&A]
gpt-engineer built me a nice codebase, but I want to improve the app it built and give it futher specifications. Is there a way that I can continue to reprompt gpt-engineer and have it understand/know the current status of the project, and how to add onto it based on my prompts?

**Comment by RareMojo** [comment]
I think a lot of us are experimenting with prompt engineering and STEP design.

In my fork, I have a review identity that utilizes memory files from a previous run. The step is just built in a logical way that pulls relevant info that would help 'resume' or 'retry'. I haven't really had much time to play with it, but it seems functional. I found that including "tests" memory or whatever you are calling it helps. 

For example, it respecs snake game, but looking at the unit tests compared to the initial project, it would fail one of those tests in real life, so it would try and implement or add the changes.

Last time I did a pull, it seemed like the main project was about to implement a respec identity and STEP. 
`--steps-config "respec"`

Anyway, I hope this helps!

**Reply by ghost** [reply]
Hey thanks for your response, I'm a bit confused on that last line of code in your message, is that a command? What should I do with that? Also, just making sure I understood what you said correctly, is follow-up with gpt-engineer after the initial prompt out of the question thus far, until Anton or one of the collaborators comes up with a solution?

**Reply by RareMojo** [reply]
I'm sorry. Yes, when you launch the main command to run gpt-engineer, there are some arguments you can make.

`Py main.py project --steps-config "default"`

For example.

It may be implemented in the main project by now. I haven't pulled since Sunday, lol. Mine does do this, though, yes, so I can't imagine it's outside the realm of possibilities as that's where I got the idea and continued from!

Backup your project and just try it.


`Py main.py project --steps-config "respec"`
